invokeai/frontend/web/src/features/parameters/hooks/useMayUpscale.ts

```diff
@@ -0,0 +1,94 @@
+import { createSelector } from '@reduxjs/toolkit';
+import { stateSelector } from 'app/store/store';
+import { useAppSelector } from 'app/store/storeHooks';
+import { defaultSelectorOptions } from 'app/store/util/defaultMemoizeOptions';
+import { useMemo } from 'react';
+import { useTranslation } from 'react-i18next';
+import { ImageDTO } from 'services/api/types';
+
+const getUpscaledPixels = (imageDTO?: ImageDTO, maxUpscalePixels?: number) => {
+  if (!imageDTO) {
+    return;
+  }
+  if (!maxUpscalePixels) {
+    return;
+  }
+  const { width, height } = imageDTO;
+  const x4 = height * 4 * width * 4;
+  const x2 = height * 2 * width * 2;
+  return { x4, x2 };
+};
+
+const getMayUpscale = (
```

Comment:
This may just be me but I keep getting confused about `mayUpscale` vs `maxUpscale`

Reply:
`mayUpscale` changed to `isAllowedToUpscale`